python/ctsm/modify_input_files/fsurdat_modifier.py

```diff
@@ -29,112 +29,382 @@ def main():
+                + " Special idealized vars ="
+                + str(ideal_list)
+            )
+        if subgrid_list.count(var) != 0:
+            abort(
+                var
+                + " is a variable handled in the subgrid section."
+                + " This should NOT be handled in the variiable list section."
+                + " Subgrid vars ="
+                + str(subgrid_list)
+            )
+        if monthly_list.count(var) != 0:
+            abort(
+                var
+                + " is a variable handled as part of the dom_pft handling."
+                + " This should NOT be handled in the variiable list section."
+                + " Monthly vars handled this way ="
+                + str(monthly_list)
+            )
+        value = get_config_value_or_array(
+            config=config, section=section, item=var, convert_to_type=float
+        )
+        varlist_settings[var] = value
+
+    return varlist_settings
+
+
+def modify_optional(
+    modify_fsurdat,
+    idealized,
+    include_nonveg,
+    max_sat_area,
+    std_elev,
+    soil_color,
+    dom_pft,
+    lai,
+    sai,
+    hgt_top,
+    hgt_bot,
+):
+    """Modify the dataset according to the optional settings"""
+
+    # Set fsurdat variables in a rectangle that could be global (default).
+    # Note that the land/ocean mask gets specified in the domain file for
+    # MCT or the ocean mesh files for NUOPC. Here the user may specify
+    # fsurdat variables inside a box but cannot change which points will
+    # run as land and which as ocean.
+    if idealized:
+        modify_fsurdat.set_idealized()  # set 2D variables
+        # set 3D and 4D variables pertaining to natural vegetation
+        modify_fsurdat.set_dom_pft(dom_pft=0, lai=[], sai=[], hgt_top=[], hgt_bot=[])
+        logger.info("idealized complete")
+
+    if max_sat_area is not None:  # overwrite "idealized" value
+        modify_fsurdat.setvar_lev0("FMAX", max_sat_area)
+        logger.info("max_sat_area complete")
+
+    if std_elev is not None:  # overwrite "idealized" value
+        modify_fsurdat.setvar_lev0("STD_ELEV", std_elev)
+        logger.info("std_elev complete")
+
+    if soil_color is not None:  # overwrite "idealized" value
+        modify_fsurdat.setvar_lev0("SOIL_COLOR", soil_color)
+        logger.info("soil_color complete")
+
+    if not include_nonveg:
+        modify_fsurdat.zero_nonveg()
+        logger.info("zero_nonveg complete")
+
+    # set_dom_pft follows zero_nonveg because it modifies PCT_NATVEG
+    # and PCT_CROP in the user-defined rectangle
+    if dom_pft is not None:
+        modify_fsurdat.set_dom_pft(
+            dom_pft=dom_pft, lai=lai, sai=sai, hgt_top=hgt_top, hgt_bot=hgt_bot
+        )
+        logger.info("dom_pft complete")
+
+
+def read_opt_config(modify_fsurdat, config, cfg_path, section):
+    """read the optional parts of the main section config file"""
+    # not required: user may set these in the .cfg file
```

Comment:
@ekluzek and I discussed this.
I propose renaming this function to show that it is operating on the optional stuff in the cfg file's first section only. The first section is `modify_fsurdat_basic_options`, so maybe name the function `read_cfg_optional_basic_opts`. 

For consistency, I suggest lengthening the docstring to say 
`Read the optional parts of the main section of the config file. The main section is called modify_fsurdat_basic_options. Users may set these optional parts but are not required to do so.`

This way the docstring incorporates the comment that's below it. If you update the docstring, then you can delete the separate comment.

Reply:
OK, I did that change.